### ISSUE TITLE
Training ImageNet will stuck at validate function

### ISSUE THREAD

**Issue by stevenyslin** [pytorch/examples#267]
Dear all,

When I use examples/imagenet/main.py to train ImageNet, the processing will stuck as following:
```
Epoch: [0][19950/20019] Time 0.544 (0.634)      Data 0.001 (0.111)      Loss 4.1225 (5.4584)    Prec@1 18.750 (7.144)   Prec@5 37.500 (18.171)
Epoch: [0][19960/20019] Time 0.965 (0.634)      Data 0.446 (0.111)      Loss 3.9474 (5.4579)    Prec@1 28.125 (7.149)   Prec@5 50.000 (18.180)
Epoch: [0][19970/20019] Time 0.517 (0.634)      Data 0.000 (0.111)      Loss 4.1201 (5.4573)    Prec@1 14.062 (7.153)   Prec@5 35.938 (18.188)
Epoch: [0][19980/20019] Time 0.719 (0.634)      Data 0.189 (0.111)      Loss 4.8537 (5.4568)    Prec@1 7.812 (7.157)    Prec@5 26.562 (18.198)
Epoch: [0][19990/20019] Time 0.526 (0.634)      Data 0.001 (0.111)      Loss 4.8936 (5.4562)    Prec@1 9.375 (7.161)    Prec@5 32.812 (18.207)
Epoch: [0][20000/20019] Time 0.837 (0.634)      Data 0.327 (0.111)      Loss 3.7931 (5.4555)    Prec@1 18.750 (7.166)   Prec@5 46.875 (18.218)
Epoch: [0][20010/20019] Time 0.512 (0.634)      Data 0.000 (0.111)      Loss 4.3824 (5.4550)    Prec@1 18.750 (7.171)   Prec@5 37.500 (18.226)
Test: [0/782]   Time 3.426 (3.426)      Loss 2.8972 (2.8972)    Prec@1 57.812 (57.812)  Prec@5 64.062 (64.062)
```
My environment as following:
pytorch version: 0.3.0.post4 (using conda install)
CUDA: 8.0.61
python: 3.5.4

And command is:
`CUDA_VISIBLE_DEVICES=0 python main.py --arch resnet101 --lr 0.1 --epochs 20 --batch-size 64 /ImageNet_database`

How could I debug this problem?
Thank you for your help.

**Comment by SsnL** [comment]
It could be IO problem. Could you add some more debug prints for validation set data loading?

**Comment by stevenyslin** [comment]
Hi SsnL,

Thank you for your response.
Maybe you are right, because when I run the script again, the problem has not appeared.
